src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -280,9 +281,11 @@ public Document GetRequiredDocument()
                 (workspace, solution) = await lspWorkspaceManager.GetLspSolutionInfoAsync(cancellationToken).ConfigureAwait(false);
             }
 
-            if (workspace is null)
+            if (workspace is null || solution is null)
             {
-                logger.LogError($"Could not find appropriate workspace for operation {workspace} on {method}");
+                logger.LogError($"Could not find appropriate workspace solution on {method}");
```

Comment:
```suggestion
                logger.LogError($"Could not find appropriate workspace or solution on {method}");
```

Or maybe just have a single string for both of these at this point.